### DISCUSSION TITLE
Should we retire the Zebra RFC process?

### DISCUSSION THREAD

**Discussion by teor2345** [ZcashFoundation/zebra#7974] [Team Decisions]
### Motivation

Some external developers have asked us if they should write Zebra RFCs. But the Zebra team itself hasn't written a RFC for 2 years.

This is because:
- RFCs were a lot of work to create, because they need a PR approval for every minor change
- A point-in-time RFC quickly becomes outdated

Instead, we have been using these docs for these reasons:
- A notion pad, google doc, HackMD, or google diagram is easier to draft and collaboratively edit
- It's easier to keep Rust module documentation up to date, because it's closer to the code
- If we need extra documentation, we just put a living document directly in the `dev` (or `user`) docs in the book

Since the unused and outdated RFC process can be confusing, let's decide if we want to retire it?
If we do, let's document how we want developers to do requirements/designs in a living `dev` book doc? 

The existing RFCs can be archived, and slowly migrated to living documents if needed.

### Outdated Specifications

If we retire RFCs, the template, CONTRIBUTING.md, the RFC section page, and the links to the RFC section should be updated:
https://github.com/ZcashFoundation/zebra/blob/main/book/src/dev/rfcs/0000-template.md

**Comment by tomekpiotrowski** [comment]
Thank you for the clarification!

Just want to add that RFCs are also mentioned in [CONTRIBUTING.md](https://github.com/ZcashFoundation/zebra/blob/main/CONTRIBUTING.md)

**Reply by teor2345** [reply]
Thanks, I've added that to the description.

**Comment by arya2** [comment]
> Should we retire the Zebra RFC process?

Yes, I don't think the process is currently worthwhile for new RFCs, and that we should retire it until that changes.

Though I do like the existing RFCs as explanations of how parts of Zebra work (though some of them may need updates), and I suspect they were very valuable in making good design decisions earlier.

**Reply by teor2345** [reply]
I think we're slowly migrating them into living documents.

I'm not saying we should delete existing RFCs, just retire the process.

**Comment by mpguerra** [comment]
I don't have any objections to retiring the RFC process. We should ask everyone else in the team to comment on this before we make a final decision.

**Comment by teor2345** [comment]
Just to confirm, I think we should retire the RFC process. But I'm open to changing my mind if there are significant drawbacks. Please add any drawbacks you can see to this discussion.

**Comment by mpguerra** [comment]
Last call for comments on this one: @upbqdn @conradoplg @natalieesk @oxarbitrage @gustavovalverde 

If no objections, I will go ahead and close this discussion with a resolution to retire the RFC process and create an issue to explain how to document requirements and designs for new zebra features.

**Reply by upbqdn** [reply]
I think we can close this discussion now, even without creating the issue.
